src/main/java/it/gov/pagopa/pu/debtpositions/service/create/receipt/CreateReceiptServiceImpl.java

```diff
@@ -58,58 +60,45 @@ public ReceiptDTO createReceipt(ReceiptWithAdditionalNodeDataDTO receiptDTO, Str
-        if (!ReceiptOriginType.RECEIPT_FILE.equals(receiptInDb.getReceiptOrigin())) {
-          throw new ConflictErrorException("Receipt having paymentReceiptId " + receiptDTO.getPaymentReceiptId() +
-            " has already been stored with origin " + receiptInDb.getReceiptOrigin() + " and id " + receiptInDb.getReceiptId());
-        }
-        logReceiptData("Updating Receipt manually imported receiptOrigin[" + receiptDTO.getReceiptOrigin() + "]", receiptDTO);
-      } else {
-        logReceiptData("Skipping Receipt already handled receiptOrigin[" + receiptDTO.getReceiptOrigin() + "]", receiptDTO);
-        // Nothing to do (neither updating data), this event has already been handled
-        return receiptInDb;
-      }
+      log.info("Receipt with paymentReceiptId[{}] already present in DB id[{}]", receiptDTO.getPaymentReceiptId(), receiptInDb.getReceiptId());
+      receiptDTO.setReceiptId(receiptInDb.getReceiptId());
+      receiptDTO.setNoPII(receiptInDb);
+      return receiptDTO;
     }
     return null;
   }
 
   private static void logReceiptData(String message, ReceiptWithAdditionalNodeDataDTO receiptDTO) {
-    log.info("{} paymentReceiptId[{}} org/nav/iud[{}/{}/{}]",
+    log.info("{} paymentReceiptId[{}] org/nav/iud[{}/{}/{}]",
       message,
       receiptDTO.getPaymentReceiptId(),
       receiptDTO.getOrgFiscalCode(),
       receiptDTO.getNoticeNumber(),
       receiptDTO.getIud());
   }
 
-  private ReceiptDTO checkIfAlreadyStored(ReceiptDTO receiptDTO) {
-    ReceiptNoPII receiptInDb = receiptNoPIIRepository.getByPaymentReceiptId(receiptDTO.getPaymentReceiptId());
-    if (receiptInDb != null) {
-      if(!receiptInDb.getReceiptOrigin().equals(receiptDTO.getReceiptOrigin())){
-        throw new ConflictErrorException("Receipt having paymentReceiptId " + receiptDTO.getPaymentReceiptId() +
-          " has already been stored with origin " + receiptInDb.getReceiptOrigin() + " and id " + receiptInDb.getReceiptId() +
-          " while the requested Receipt has origin " + receiptDTO.getReceiptOrigin());
-      }
-      log.info("Receipt with paymentReceiptId[{}] already present in DB id[{}]", receiptDTO.getPaymentReceiptId(), receiptInDb.getReceiptId());
-      receiptDTO.setReceiptId(receiptInDb.getReceiptId());
-      receiptDTO.setNoPII(receiptInDb);
-      return receiptDTO;
+  private boolean shouldUpdateExistingReceipt(ReceiptWithAdditionalNodeDataDTO receiptDTO, ReceiptDTO existingReceipt) {
+    if (StringUtils.isEmpty(receiptDTO.getIud())) {
+      log.info("Skipping update: receipt already exists and is not a manual import");
+      return false;
     }
-    return null;
-  }
 
-  private boolean isManualImport(ReceiptWithAdditionalNodeDataDTO receiptDTO) {
-    return !StringUtils.isEmpty(receiptDTO.getIud());
+    if (ReceiptOriginType.RECEIPT_FILE.equals(existingReceipt.getReceiptOrigin())) {
+      logReceiptData("Updating Receipt manually imported with technical origin", receiptDTO);
```

Comment:
this sentence is ambigous, it seems that the stored receipt is a manual import
`Updating receipt: the input is a manual import and the stored receipt has technical origin: {}`
let's log the stored origin